secboot/keymgr/keymgr_luks2.go

```diff
@@ -42,19 +45,44 @@ const (
 
 var (
 	sbGetDiskUnlockKeyFromKernel = sb.GetDiskUnlockKeyFromKernel
+	disksDevlinks                = disks.Devlinks
 )
 
 func getEncryptionKeyFromUserKeyring(dev string) ([]byte, error) {
 	const remove = false
 	const defaultPrefix = "ubuntu-fde"
+
 	// note this is the unlock key, which can be either the main key which
 	// was unsealed, or the recovery key, in which case some operations may
 	// not make sense
-	currKey, err := sbGetDiskUnlockKeyFromKernel(defaultPrefix, dev, remove)
-	if err != nil {
+	p, err := sbGetDiskUnlockKeyFromKernel(defaultPrefix, dev, remove)
+	if err == nil {
+		return p, nil
+	}
+	if !errors.Is(err, sb.ErrKernelKeyNotFound) {
 		return nil, fmt.Errorf("cannot obtain current unlock key for %v: %v", dev, err)
 	}
-	return currKey, err
+
+	// Old kernels will use "by-partuuid" symlinks. So let's
+	// look at all the symlinks of the device.
+	devlinks, errDevlinks := disksDevlinks(dev)
+	if errDevlinks != nil {
+		return nil, err
+	}
+	var errDevlink error
```

Comment:
I would log something here as well, such that in the event we need to debug something from the logs there's some context to the error:
```
logger.Noticef("unlock key for device %s not found in kernel keyring, trying compatibility symlinks: %v",
                    dev, devlinks)
```

Reply:
It is a lot of symlinks, and most of them will not be tried. So I put that only for the ones we try.

Reply:
I also had to initialize the logger in snap-fde-keymgr, it was missing.